### DISCUSSION TITLE
Launching Adobe AIR runtime file causes error to pop up

### DISCUSSION THREAD

**Discussion by LinhconT** [airsdk/Adobe-Runtime-Support#2592] [Q&A]
![error](https://user-images.githubusercontent.com/48224426/231694345-4286ef40-8d1f-46b3-90a5-b8f9003170d9.PNG)
Every time I try to launch the Adobe AIR runtime file from the Harman download, this message pops up. I tried searching for fixes online, but nothing really seems to help. Any help would be appreciated!

**Comment by ajwfrost** [comment]
Do you have any strict anti-virus that may be killing the executable? I think it should be properly signed though... which version of the runtime have you got?

You could try opening the AdobeAIR.exe file using a zip manager and extracting the files manually into a folder where you can then run the "Adobe AIR Installer.exe" manually, and see if there's any more clues?

**Reply by LinhconT** [reply]
I'm using the default Windows Anti Virtus, so it really shouldn't be killing or quarantining the installer. The version of the runtime I have installed is the latest one from the Harman website, so v50.2.1.1. Fresh installs of the runtime is caught by Defender SmartScreen, but I am able to run the file with this message still popping up.

I have tried installing the SDK, but no decent results from there.

![error2](https://user-images.githubusercontent.com/48224426/231920130-54053f1d-c939-4061-bd61-2fa32b0fbae0.PNG)
After extracting the AdobeAIR.exe contents to another folder, I have looked for the Adobe AIR Installer.exe file, but running it gives me an error saying that the installer requires a version of Adobe AIR that cannot be found.

![error3](https://user-images.githubusercontent.com/48224426/231920475-c357785a-75bc-4505-878d-2609739f8c43.PNG)
I tried running the setup.msi file, but it gives me an alert saying that it could not access the network location, and pressing cancel just repeats the same message before closing.

Digging into the Resources folder, there is another Adobe AIR Installer executable file, but trying to run it just simply says that my computer cannot run it and "I need to contact the software developer".

Edit - I did some more digging, and I eventually just decided to copy paste the Adobe AIR contents into the AIR files that were created in my AppData folder, and I just found out that whenever you run the AdobeAIR.exe file, it keeps creating folders in the AppData temp folder.

**Reply by ajwfrost** [reply]
So the first one there is running the "AIR Application Installer", and yes that shouldn't work unless the AIR shared runtime is installed.
The MSI likewise isn't meant to be run directly, I recall seeing that message myself though in some scenarios - I think it may be a similar reason though in terms of the runtime library not loading.

But the "AdobeAIR.exe" is a self-extracting archive that also includes another file "Adobe AIR Installer.exe" alongside those set-up files...
![image](https://user-images.githubusercontent.com/6400804/231979593-e1eb6765-59ec-4baa-8091-01516fffcd27.png)


And you're also missing the actual AIR library ("Adobe AIR.dll") that should be alongside the "Adobe AIR Application Installer.exe" files, unless you've got DLLs set as hidden as system files in your explorer options..?

So are you able to check that in a zip manager, and see what's happening to it when you extract it? I've just checked and everything is code-signed with EV certificates so it shouldn't be caught by any AV programs, least of all Microsoft's own one..

thanks

**Reply by LinhconT** [reply]
Apparently the Adobe AIR.dll file is actually corrupt when trying to extract it through WinRAR. It also lists a lot of other files as being corrupt too.
![corrupt](https://user-images.githubusercontent.com/48224426/232171696-a4d7650b-c840-4e23-a190-e1598a71de8c.PNG)

I can't seem to find an older version of the runtime on the Harman website, so I may have to resort to waiting for someone to send me a runtime that isn't corrupted.

**Reply by ajwfrost** [reply]
Okay so I've just clicked on the link on the website again, i.e. https://airsdk.harman.com/assets/downloads/AdobeAIR.exe
Quick sha-256 checksum check -> ac6a65976b583b64c11d4592037f25f8115e755e74b31a55c816627a0449b547 
I don't have WinRAR but in 7-zip:
![image](https://user-images.githubusercontent.com/6400804/232358325-8c3056de-5f0b-43de-b46e-c9dd6362c4b4.png)

And it extracts and runs fine.

Can you double-check the checksum of the file you downloaded? I'm wondering if this is just some slight corruption in the download..

thanks

**Reply by LinhconT** [reply]
The link that you provided is perfectly fine, and now Adobe AIR has installed properly onto my computer.

I'm not too sure how to check the checksum of the file I downloaded, but all should be good now. Thanks for the help!

**Reply by ajwfrost** [reply]
Great, thanks -> I've got 'cygwin' installed that gives me an "shasum" application, the same exists on macs and linux. I'm sure there are tools online for it... or you can write one in ActionScript :-) https://airsdk.dev/reference/actionscript/3.0/air/security/Digest.html

thanks
